src/emmy/complex/impl.cljc

```diff
@@ -0,0 +1,616 @@
+#_"SPDX-License-Identifier: GPL-3.0"
+
+;; /**
+;;  * @license Complex.js v2.1.1 12/05/2020
+;;  *
+;;  * Copyright (c) 2020, Robert Eisele (robert@xarg.org)
+;;  * Dual licensed under the MIT or GPL Version 2 licenses.
+;;  **/
+
+(ns emmy.complex.impl
+  "This namespace provides the primitive implementation of complex
+   arithmetic for Emmy. The authors are indebted to Robert Eisele's
+   `Complex.js` implementation from which we have borrowed freely and
+   which is licensed as follows:
+
+   * Copyright (c) 2020, Robert Eisele (robert@xarg.org)
+   * Dual licensed under the MIT or GPL Version 2 licenses."
+  (:refer-clojure :exclude [abs zero?])
+  (:require [emmy.generic :as g]
+            [emmy.util :as u]
+            [emmy.value :as v]))
+
+(declare equal?)
+
+(deftype Complex [re im]
+  v/IKind
+  (kind [_] :emmy.complex/complex)
+
+  v/Numerical
+  (numerical? [_] true)
+
+  #?@(:clj [Object
+            (equals [a b] (equal? a b))]
+      :cljs [IEquiv
+             (-equiv [a b] (equal? a b))
+
+             IPrintWithWriter
+             (-pr-writer
+              [z writer _]
+              (write-all
+               writer
+               "#emmy/complex "
+               (str [(.-re z) (.-im z)])))]))
```

Comment:
Adding this should get you the same feature for the clj side if it's not somewhere else already:

```clojure
#?(:clj
   (defmethod print-method Complex
     [^Complex z ^java.io.Writer w]
     (.write w "#emmy/complex ")
     (.write w (pr-str [(.-re z) (.-im z)]))))
```